File: MotivApp/app/src/main/java/com/motiv/piotr/DashboardActivity.java
package com.motiv.piotr;

import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.drawerlayout.widget.DrawerLayout;
import com.google.android.material.navigation.NavigationView;
import com.motiv.piotr.dao.DaoRepository;
import com.motiv.piotr.dao.DaoRepositoryFactory;
import com.motiv.piotr.dao.LocalStorage;
import com.squareup.picasso.Picasso;

public class DashboardActivity extends AppCompatActivity implements DashboardActivityContract.View {

    private com.motiv.piotr.User loggedUser;
    private DashboardActivityContract.Presenter presenter;
    private UsersListAdapter usersListAdapter;
    private PostsListAdapter postsListAdapter;
    private PhotosPagerAdapter photosPagerAdapter;
    private FragmentsPagerAdapter fragmentsPagerAdapter;
    private GoRestApi goRestApi;
    private DaoRepository daoRepository;
    private LocalStorage localStorage;
    private NavigationController navigationController;
    private DrawerLayout drawerlayout00;
    private NavigationView navigationview11;
    private LinearLayout headerlinearlayout00;
    private ImageView headerimageview10;
    private TextView headertextview11;

    @Override
    protected void onCreate(@Nullable android.os.Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.dashboardactivity);

        loggedUser = com.motiv.piotr.User.fromJson(getIntent().getStringExtra("loggedUser"));

        usersListAdapter = new UsersListAdapter();
        postsListAdapter = new PostsListAdapter();
        photosPagerAdapter = new PhotosPagerAdapter();
        fragmentsPagerAdapter =
                new FragmentsPagerAdapter(DashboardActivity.this.getSupportFragmentManager());
        daoRepository = DaoRepositoryFactory.getInstance(DashboardActivity.this);
        localStorage = LocalStorage.getInstance(DashboardActivity.this);
        navigationController = new NavigationController(DashboardActivity.this);
        goRestApi = GoRestApiFactory.getInstance(localStorage);
        drawerlayout00 = (DrawerLayout) findViewById(R.id.drawerlayout00);
        navigationview11 = (NavigationView) findViewById(R.id.navigationview11);
        headerlinearlayout00 =
                (LinearLayout) navigationview11.getHeaderView(0).findViewById(R.id.linearlayout00);
        headerimageview10 =
                (ImageView) navigationview11.getHeaderView(0).findViewById(R.id.imageview10);
        headertextview11 =
                (TextView) navigationview11.getHeaderView(0).findViewById(R.id.textview11);

        presenter =
                new DashboardActivityPresenter(
                        DashboardActivity.this, goRestApi, daoRepository, localStorage);

        navigationview11.setNavigationItemSelectedListener(
                new com.google.android.material.navigation.NavigationView
                        .OnNavigationItemSelectedListener() {
                    @Override
                    public boolean onNavigationItemSelected(android.view.MenuItem argument0) {
                        presenter.eloonNavigationItemSelected(argument0);

                        return false;
                    }
                });
        headertextview11.setText(loggedUser.getFirst_name());

        Picasso.with(DashboardActivity.this)
                .load(loggedUser.getLinks().getAvatar().getHref())
                .into(headerimageview10);
        ;
        headerlinearlayout00.setOnClickListener(
                new android.view.View.OnClickListener() {
                    @Override
                    public void onClick(android.view.View argument0) {
                        presenter.eloonClick(argument0);
                    }
                });
    }

    @Override
    public void navigationControllerstartDestinationById(int arg0) {
        navigationController.startDestinationById(arg0);
    }

    @Override
    public void navigationControllerstartUserDetailActivity(com.motiv.piotr.User arg0) {
        navigationController.startUserDetailActivity(arg0);
    }

    @Override
    public com.motiv.piotr.User getloggedUser() {
        return loggedUser;
    }
}

File: MotivApp/app/src/main/java/com/motiv/piotr/DashboardActivity.kt
 
package com.motiv.piotr
import android.os.Bundle
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.drawerlayout.widget.DrawerLayout
import com.google.android.material.navigation.NavigationView
import com.motiv.piotr.dao.DaoRepository
import com.motiv.piotr.dao.LocalStorage
import com.motiv.piotr.databinding.DashboardactivityBinding
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.dashboardactivity.*

public class DashboardActivity : AppCompatActivity(), DashboardActivityContract.View {

    private lateinit var dashboardactivityBinding: DashboardactivityBinding

    private lateinit var loggedUser: com.motiv.piotr.User

    private lateinit var presenter: DashboardActivityContract.Presenter

    private lateinit var usersListAdapter: UsersListAdapter

    private lateinit var postsListAdapter: PostsListAdapter

    private lateinit var photosPagerAdapter: PhotosPagerAdapter

    private lateinit var fragmentsPagerAdapter: FragmentsPagerAdapter

    private lateinit var goRestApi: GoRestApi

    private var daoRepository: DaoRepository = DaoRepositoryFactory.getInstance(this@DashboardActivity)

    private lateinit var localStorage: LocalStorage

    private lateinit var navigationController: NavigationController

    private lateinit var drawerlayout00: DrawerLayout

    private lateinit var navigationview11: NavigationView

    private lateinit var headerlinearlayout00: LinearLayout

    private lateinit var headerimageview10: ImageView

    private lateinit var headertextview11: TextView

    override fun onCreate(savedInstanceState: android.os.Bundle?) {
        super.onCreate(savedInstanceState)
        dashboardactivityBinding = DataBindingUtil.setContentView(this, R.layout.dashboardactivity)

        loggedUser = com.motiv.piotr.User.fromJson(getIntent().getStringExtra("loggedUser"))

        usersListAdapter = UsersListAdapter()
        postsListAdapter = PostsListAdapter()
        photosPagerAdapter = PhotosPagerAdapter()
        fragmentsPagerAdapter = FragmentsPagerAdapter(this@DashboardActivity.getSupportFragmentManager())
        localStorage = LocalStorage.getInstance(this@DashboardActivity)
        navigationController = NavigationController(this@DashboardActivity)
        goRestApi = GoRestApiFactory.getInstance(localStorage)
        drawerlayout00 = dashboardactivityBinding.drawerlayout00
        navigationview11 = dashboardactivityBinding.navigationview11
        headerlinearlayout00 = navigationview11.getHeaderView(0).findViewById<LinearLayout>(R.id.linearlayout00)
        headerimageview10 = navigationview11.getHeaderView(0).findViewById<ImageView>(R.id.imageview10)
        headertextview11 = navigationview11.getHeaderView(0).findViewById<TextView>(R.id.textview11)

        presenter = DashboardActivityPresenter(this@DashboardActivity, goRestApi, daoRepository, localStorage)

        navigationview11.setNavigationItemSelectedListener(object : com.google.android.material.navigation.NavigationView.OnNavigationItemSelectedListener {
            override fun onNavigationItemSelected(argument0: android.view.MenuItem): Boolean {
                presenter.eloonNavigationItemSelected(argument0)

                return false
            } 
        })
        headertextview11.setText(loggedUser.getFirst_name())

        Picasso.with(this@DashboardActivity).load(loggedUser.getLinks().getAvatar().getHref()).into(headerimageview10)
        headerlinearlayout00.setOnClickListener(object : android.view.View.OnClickListener {
            override fun onClick(argument0: android.view.View) {
                presenter.eloonClick(argument0)
            } 
        })
    } override fun navigationControllerstartDestinationById(arg0: Int) {
        navigationController.startDestinationById(arg0)
    } override fun navigationControllerstartUserDetailActivity(arg0: com.motiv.piotr.User) {
        navigationController.startUserDetailActivity(arg0)
    } override fun getloggedUser(): com.motiv.piotr.User {
        return loggedUser
    }
}
